addons/hr_holidays/tests/test_company_leave.py

```diff
@@ -0,0 +1,329 @@
+            'request_date_from_period': 'am',
+
+        })
+        leave._onchange_request_parameters()
+
+        company_leave = self.env['hr.leave'].create({
+            'name': 'Bank Holiday',
+            'holiday_type': 'company',
+            'mode_company_id': self.company.id,
+            'holiday_status_id': self.bank_holiday.id,
+            'date_from': date(2020, 1, 7),
+            'request_date_from': date(2020, 1, 7),
+            'date_to': date(2020, 1, 7),
+            'request_date_to': date(2020, 1, 7),
+            'number_of_days': 1,
+        })
+        company_leave._onchange_request_parameters()
+
+        company_leave.action_validate()
+
+        all_leaves = self.env['hr.leave'].search([('employee_id', '=', self.employee.id)], order='id')
+        self.assertEqual(len(all_leaves), 2)
+        # Original Leave
+        self.assertEqual(leave.state, 'refuse')
+        # Company Leave
+        self.assertEqual(all_leaves[1].date_from, datetime(2020, 1, 7, 7, 0))
+        self.assertEqual(all_leaves[1].date_to, datetime(2020, 1, 7, 16, 0))
+        self.assertEqual(all_leaves[1].number_of_days, 1)
+        self.assertEqual(all_leaves[1].state, 'validate')
+
+    def test_leave_whole_company_04(self):
+        # TEST CASE 4: Leaves taken in days. Take a 1 days leave
+        # Add a company leave on the same day
+        # Check that leave is refused
+        self.paid_time_off.request_unit = 'day'
+
+        leave = self.env['hr.leave'].create({
+            'name': 'Hol11',
+            'employee_id': self.employee.id,
+            'holiday_status_id': self.paid_time_off.id,
+            'request_date_from': date(2020, 1, 9),
+            'request_date_to': date(2020, 1, 9),
+            'number_of_days': 1,
+
+        })
+        leave._onchange_request_parameters()
+
+        company_leave = self.env['hr.leave'].create({
+            'name': 'Bank Holiday',
+            'holiday_type': 'company',
+            'mode_company_id': self.company.id,
+            'holiday_status_id': self.bank_holiday.id,
+            'date_from': date(2020, 1, 9),
+            'request_date_from': date(2020, 1, 9),
+            'date_to': date(2020, 1, 9),
+            'request_date_to': date(2020, 1, 9),
+            'number_of_days': 1,
+        })
+        company_leave._onchange_request_parameters()
+
+        company_leave.action_validate()
+
+        all_leaves = self.env['hr.leave'].search([('employee_id', '=', self.employee.id)], order='id')
+        self.assertEqual(len(all_leaves), 2)
+        # Original Leave
+        self.assertEqual(leave.state, 'refuse')
+        # Company Leave
+        self.assertEqual(all_leaves[1].date_from, datetime(2020, 1, 9, 7, 0))
+        self.assertEqual(all_leaves[1].date_to, datetime(2020, 1, 9, 16, 0))
+        self.assertEqual(all_leaves[1].number_of_days, 1)
+        self.assertEqual(all_leaves[1].state, 'validate')
+
+    def test_leave_whole_company_05(self):
```

Comment:
Is this test useful? I have the impression that it tests the same thing as the previous test...

Reply:
You're probably right ! :)

I thought there was a bug, but the test didn't expose it actually.